### PR TITLE
Fix up failing tests in Internet Explorer

### PR DESCRIPTION
**Pull request by bartvde**
A lot of test cases currently fail in Internet Explorer.

This is a WIP on fixing them. Will update the description once I'm done.

### CODE REVIEW

**Comment by ahocevar** [comment]
Are you also testing the examples manually to see if there is something in the library that needs to be fixed? Do you want me to help? I think we should fully support IE10+.

**Comment by bartvde** [comment]
Not as yet @ahocevar but that might make sense to do as well after I finish up the unit tests.
Have we discarded IE9 as supported? I thought we still supported IE9?

**Comment by bartvde** [comment]
do you remember what was the case with ```ol.source.Raster``` and IE? IIRC this is unsupported / can't be made to work in IE?

**Comment by bartvde** [comment]
on second runs strange things happen, like pixelworks error for ol.source.TileJSON, maybe due to test cleanup?

screen shot 2015-11-23 at 11 04 44

**Comment by bartvde** [comment]
if I take out the ```ol.source.Raster``` test spec, all is fine now on all runs, so something is going on there or with pixelworks ext

**Comment by bartvde** [comment]
```ol.format.MVT``` uses ArrayBuffer so needs IE10+

**Comment by bartvde** [comment]
This seems to be the culprit test case:

```
    itNoPhantom('gets called before operations are run', function(done) {

      var count = 0;
      raster.setOperation(function(inputs) {
        ++count;
        return inputs[0];
      });

      raster.on('beforeoperations', function(event) {
        expect(count).to.equal(0);
        expect(!!event).to.be(true);
        expect(event.extent).to.be.an('array');
        expect(event.resolution).to.be.a('number');
        expect(event.data).to.be.an('object');
        done();
      });

      var view = map.getView();
      view.setCenter([0, 0]);
      view.setZoom(0);

    });
```

**Comment by ahocevar** [comment]
I'm looking into the raster issue now.

File: test/spec/ol/format/esrijsonformat.test.js [hunk 1]
```diff
@@ -304,15 +304,15 @@ describe('ol.format.EsriJSON', function() {
               -10579560.16462974, 4716567.373073828]))
             .to.be(true);
 
-        var last = result[305];
+        var last = result[8];
```
Comment (ahocevar) [inline test/spec/ol/format/esrijsonformat.test.js]: Now `last` is not a good name any more. `feature`?

Comment (bartvde) [inline test/spec/ol/format/esrijsonformat.test.js]: It's still the last feature, I just made the file smaller because it was very slow parsing in IE

Comment (ahocevar) [inline test/spec/ol/format/esrijsonformat.test.js]: Ah ok. You didn't push the data change.

Comment (bartvde) [inline test/spec/ol/format/esrijsonformat.test.js]: I did, it's here: test/spec/ol/format/esrijson/ksfields.json just on top of this

Comment (ahocevar) [inline test/spec/ol/format/esrijsonformat.test.js]: Oh sorry, you did. It's just collapsed in the diff.

**Comment by bartvde** [comment]
The example template has a minor issue in IE11, the select combo option only shows up on hover, not by default. Seems to be caused by color: inherit from the bootstrap css.

screen shot 2015-11-23 at 17 34 48
screen shot 2015-11-23 at 17 34 43
screen shot 2015-11-23 at 17 34 32

**Comment by bartvde** [comment]
Also the logo is stretched in IE, and I think we saw this before?

screen shot 2015-11-23 at 17 37 23

**Comment by tschaub** [comment]
The raster source uses the `ImageData` constructor.  I'd suggest not running the raster tests in IE for now.  A workaround for the lack of an `ImageData` constructor in IE could go in the pixelworks lib.

**Comment by bartvde** [comment]
@ahocevar your fix worked perfectly, and is in line with what @tschaub said. I applied the same fix for ol.format.MVT and IE9.

**Comment by bartvde** [comment]
All tests now pass in IE9, IE10 and IE11

**Comment by ahocevar** [comment]
Some commit messages do not comply with the rules, you may want to split them into title and description. Looks good otherwise.

**Pull request merged**